src/codec.rs

```diff
@@ -150,10 +156,23 @@ impl From<std::io::Error> for Error {
 	}
 }
 
+/// Wrapper that implements Input for any `Read` and `Seek` type.
+#[cfg(feature = "std")]
+pub struct IoReader<R: std::io::Read + std::io::Seek>(pub R);
+
 #[cfg(feature = "std")]
-impl<R: std::io::Read> Input for R {
+impl<R: std::io::Read + std::io::Seek> Input for IoReader<R> {
+	fn remaining_len(&mut self) -> Result<usize, Error> {
+		use std::convert::TryInto;
+
+		let len = self.0.seek(std::io::SeekFrom::End(0))?
```

Comment:
This is not correct. You need to implement it as shown here: https://doc.rust-lang.org/src/std/io/mod.rs.html#1398

Reply:
fixed